connector-common-secrets-loader/src/main/java/com/redhat/cloud/notifications/connector/secrets/SecretsLoader.java

```diff
@@ -0,0 +1,75 @@
+package com.redhat.cloud.notifications.connector.secrets;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import io.quarkus.logging.Log;
+import io.quarkus.runtime.configuration.ProfileManager;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.ORG_ID;
+import static com.redhat.cloud.notifications.connector.secrets.SecretsExchangeProperty.SECRET_ID;
+import static com.redhat.cloud.notifications.connector.secrets.SecretsExchangeProperty.SECRET_PASSWORD;
+import static com.redhat.cloud.notifications.connector.secrets.SecretsExchangeProperty.SECRET_USERNAME;
+import static io.quarkus.runtime.LaunchMode.TEST;
+
+@ApplicationScoped
+public class SecretsLoader implements Processor {
+
+    private static final String SECRETS_LOADER_ENABLED = "notifications.connector.secrets-loader.enabled";
+    private static final String SOURCES_API_PSK = "notifications.connector.secrets-loader.sources-api-psk";
+    private static final String SOURCES_TIMER = "sources.get.secret.request";
+
+    @ConfigProperty(name = SECRETS_LOADER_ENABLED, defaultValue = "false")
+    boolean secretsLoaderEnabled;
+
+    @ConfigProperty(name = SOURCES_API_PSK, defaultValue = "development-value-123")
+    String sourcesApiPsk;
+
+    @Inject
+    MeterRegistry meterRegistry;
+
+    @Inject
+    @RestClient
+    SourcesClient sourcesClient;
+
+    @Override
+    public void process(Exchange exchange) {
+        if (secretsLoaderEnabled) {
+            Long secretId = exchange.getProperty(SECRET_ID, Long.class);
+            if (secretId != null) {
+                String orgId = exchange.getProperty(ORG_ID, String.class);
+
+                // TODO Lower the log level after the testing phase.
+                Log.infof("Calling Sources to retrieve a secret [orgId=%s, secretId=%d]", orgId, secretId);
+
+                Timer.Sample timer = Timer.start(meterRegistry);
+                Secret secret = sourcesClient.getById(orgId, sourcesApiPsk, secretId);
```

Comment:
You didn't added any retry policy because it will be handled by usual retry on exception management on connectors, am I right ?

Reply:
There's a retry policy in `SourcesClient`.

Reply:
Shouldn't we remove it on sources client to avoid retires or retries ? (could be done later)

Reply:
Yes, we could change that later. However, Sources is an internal and stable service so the number of retries should be very low.